src/cursor_manager.ts

```diff
@@ -61,8 +61,17 @@ export class CursorManager implements Disposable {
      * cursor updates while entering insert mode and insert mode commands. Thus, when those events occur,
      * this flag is used to disable ignoring the update. This is set to true when entering insert
      * mode or running insert mode command, and set to false before document updates in insert mode.
+     *
+     * The flag corresponds to each `TextEditor`
      */
-    public wantInsertCursorUpdate = true;
+    private _wantInsertCursorUpdate: WeakMap<TextEditor, boolean> = new WeakMap();
+    public wantInsertCursorUpdate = (editor: TextEditor) => this._wantInsertCursorUpdate.get(editor) ?? false;
```

Comment:
nit: is there any reason for these not to be methods?

Reply:
Personal preference, looks more compact. 😹